subgraphs/inverse-finance/src/common/getters.ts

```diff
@@ -0,0 +1,398 @@
+  LendingProtocol,
+  Market,
+  MarketDailySnapshot,
+  MarketHourlySnapshot,
+  UsageMetricsDailySnapshot,
+  UsageMetricsHourlySnapshot,
+  FinancialsDailySnapshot,
+} from "../../generated/schema";
+import {
+  Network,
+  BIGDECIMAL_ZERO,
+  BIGINT_ZERO,
+  INT_ZERO,
+  FACTORY_ADDRESS,
+  INV_ADDRESS,
+  ZERO_ADDRESS,
+  ProtocolType,
+  LendingType,
+  RiskType,
+  MANTISSA_DECIMALS,
+  RewardTokenType,
+  InterestRateType,
+  InterestRateSide,
+  SECONDS_PER_HOUR,
+  SECONDS_PER_DAY,
+} from "../common/constants";
+
+export function getOrCreateToken(cToken: Address): Token {
+  let tokenId: string = cToken.toHexString();
+  let token = Token.load(tokenId);
+
+  if (token == null) {
+    token = new Token(tokenId);
+
+    let contract = CErc20.bind(cToken);
+    token.name = contract.name();
+    token.symbol = contract.symbol();
+    token.decimals = contract.decimals();
+
+    token.save();
+  }
+  return token;
+}
+
+export function getOrCreateUnderlyingToken(cToken: Address): Token {
+  // use default for cETH, which has no underlying
+  let tokenId = ZERO_ADDRESS;
+  let name = "Ether";
+  let symbol = "ETH";
+  let decimals = 18;
+
+  //even if the underlying token is not always a CErc20,
+  // it should work for the purpose of getting name, symbol, & decimals
+  let cTokenContract = CErc20.bind(cToken);
+  let tryUnderlyingTokenAddr = cTokenContract.try_underlying();
+  if (!tryUnderlyingTokenAddr.reverted) {
+    let tokenId = tryUnderlyingTokenAddr.value.toHexString();
+    let underlyingTokenContract = ERC20.bind(tryUnderlyingTokenAddr.value);
+    let name = underlyingTokenContract.name();
+    let symbol = underlyingTokenContract.symbol();
+    let decimals = underlyingTokenContract.decimals();
+  }
+
+  let token = Token.load(tokenId);
+
+  if (token == null) {
+    token = new Token(tokenId);
+    token.name = name;
+    token.symbol = symbol;
+    token.decimals = decimals;
+
+    token.save();
+  }
+  return token;
+}
+
+export function getUnderlyingTokenPrice(cToken: Address): BigDecimal {
+  let factoryContract = Factory.bind(Address.fromString(FACTORY_ADDRESS));
+  let oracleAddress = factoryContract.oracle() as Address;
+  let oracleContract = PriceOracle.bind(oracleAddress);
+  let underlyingPrice = oracleContract
+    .getUnderlyingPrice(cToken)
+    .toBigDecimal()
+    .div(decimalsToBigDecimal(MANTISSA_DECIMALS));
+
+  return underlyingPrice;
+}
+
+export function getUnderlyingTokenPricePerAmount(cToken: Address): BigDecimal {
+  //return price of 1 underlying token
+  let underlyingPrice = getUnderlyingTokenPrice(cToken);
+  let decimals = getOrCreateUnderlyingToken(cToken).decimals;
+  let denominator = decimalsToBigDecimal(decimals);
+  return underlyingPrice.div(denominator);
+}
+
+export function getOrCreateProtocol(): LendingProtocol {
+  let protocol = LendingProtocol.load(FACTORY_ADDRESS);
+
+  if (!protocol) {
+    protocol = new LendingProtocol(FACTORY_ADDRESS);
+    protocol.name = "Inverse Finance v1";
```

Comment:
You can remove the v1 since there is only one version and it's unclear if there is going to be a v2.